saleor/graphql/attribute/utils.py

```diff
@@ -393,6 +470,23 @@ def _pre_save_swatch_value(
         if not attr_values.swatch:
             return tuple()
 
+        attr_value = attr_values.swatch.value
+        external_ref = attr_values.swatch.external_reference
+
+        if external_ref and attr_value:
+            value = cls._create_value_instance(attribute, attr_value, external_ref)
```

Comment:
In `_pre_save_X_value` methods we are creating the instances. It could be misleading. We should consider doing it in a separate method. Also `_create_value_instance` should be handled with `bulk`. Any `attribute_models.AttributeValue.objects.get(` should be replaced by single `.filter(` call that will fetch all data instead of the record related to the one object.
Keep in mind that, without this, any usage of the Attribute mutation with big amount of attributes to add could produce huge number of unneeded DB queries.